apps/bestofjs-nextjs/src/app/api/og/projects/route.tsx

```diff
@@ -0,0 +1,96 @@
+import { formatNumber } from "@/helpers/numbers";
+import { TagIcon } from "@/components/core";
+import {
+  ProjectPageSearchParams,
+  parseSearchParams,
+} from "@/components/project-list/navigation-state";
+import {
+  SortOption,
+  SortOptionKey,
+  sortOrderOptionsByKey,
+} from "@/components/project-list/sort-order-options";
+import { api } from "@/server/api-remote-json";
+import { Box, generateImageResponse, mutedColor } from "@/app/api/og/og-utils";
+
+import { ImageLayout } from "../og-image-layout";
+import { ProjectRow } from "../og-utils";
+
+export const runtime = "edge";
+
+export async function GET(_req: Request) {
+  const { tags, query, sort, page, limit } = useSearchParams(_req.url);
+  const sortOption = getSortOption(sort);
+  const { projects, total } = await api.projects.findProjects({
+    criteria: tags.length > 0 ? { tags: { $all: tags } } : {},
+    query,
+    sort: sortOption.sort,
+    skip: limit * (page - 1),
+    limit,
```

Comment:
The limit should be set to `3` and we should ignore the `limit` from the query string.